share/spack/shell.nix

```diff
@@ -0,0 +1,16 @@
+{ pkgs ? (import <nixpkgs> {}) }:
+
+with pkgs; stdenv.mkDerivation {
+
+  buildInputs = [
```

Comment:
A `nix-shell --pure ...` would be the ultimate test to make sure this contains everything that's needed, but I'd be happy even with the current partial list here :smile:

If you're up for it https://spack.readthedocs.io/en/latest/getting_started.html#id8 and https://github.com/spack/spack/blob/7d99fbcafd4720babe2a0d325df099ed9056f5e1/var/spack/repos/builtin/packages/spack/package.py#L37-L80 will probably contain everything that's needed (except compilers).